lms/lib/courseware_search/lms_result_processor.py

```diff
@@ -73,7 +73,14 @@ def get_display_name(category, item_id):
 
         def get_position_name(section, position):
             """ helper to fetch name corresponding to the position therein """
-            pos = int(position)
+            if position is None:
+                return None
+
+            try:
+                pos = int(position.split('_', 1)[0])
+            except (ValueError, TypeError):
+                pos = 1
```

Comment:
Code duplication - if this is preserved, it should likely be moved to a separate function.